src/commands/sdk.ts

```diff
@@ -83,6 +84,54 @@ export class SdkCommands {
 				DartHoverProvider.clearPackageMapCaches();
 			}
 		}));
+		context.subscriptions.push(vs.commands.registerCommand("flutter.screenshot", async (uri) => {
+			let shouldNotify = false;
+			if (!uri || !(uri instanceof Uri)) {
+				// If there is no path for this session, get it from config
+				// If there is already a path set for this session, but the config differs
+				// set the session path to the config path and conrtinue execution. 
+				if (!this.flutterScreenshotPath) {
+					this.flutterScreenshotPath = config.flutterScreenshotPath;
+					shouldNotify = true;
+				} else if (config.flutterScreenshotPath && (this.flutterScreenshotPath !== config.flutterScreenshotPath)) {
+					this.flutterScreenshotPath = config.flutterScreenshotPath;
+					shouldNotify = true;
+				}
+
+				// If path is still empty it will bring up the folder selector.
+				if (!this.flutterScreenshotPath) {
+					const selectedFolder =
+						await window.showOpenDialog({ canSelectFolders: true, canSelectMany: false, openLabel: "Select folder to save screenshot" });
+					if (selectedFolder && selectedFolder.length > 0) {
+						// Set variable to selected path. This allows prompting the user only once.
+						this.flutterScreenshotPath = selectedFolder[0].path;
+						shouldNotify = true;
+					} else {
+						// Do nothing if the user cancelled the folder selection.
+						return;
+					}
+				}
+
+				// 	// if folder doesn't exist, create it
+				if (!fs.existsSync(this.flutterScreenshotPath)) {
+					fs.mkdirSync(this.flutterScreenshotPath);
+				}
+			}
+
+			const screenshotResult = this.runFlutterInFolder(this.flutterScreenshotPath, ["screenshot"], "screenshot");
+
+			screenshotResult.then(async (_) => {
+				if (shouldNotify) {
+					const res = await vs.window.showInformationMessage("Screenshot saved at " + this.flutterScreenshotPath.toString(), "Open folder");
+					if (res) {
+						const uri = vs.Uri.parse(this.flutterScreenshotPath);
+						// TODO: open folder in file explorer.
```

Comment:
We should either implement this, or remove the button.

The only way I can see to do this is maybe with the `revealFileInOS` command but I don't know if that's intended to be a public API (though it' used by TypeScript). @mjbvz is it possible/OK for us to use this?

https://github.com/Microsoft/vscode/blob/9fb32293774874f2c93da51cd83395c02d230e5c/extensions/typescript-language-features/src/typescriptServiceClient.ts#L521

Reply:
Not an official extension API but I think it should be ok to use since we expose it for keybindings